lib/internal/modules/esm/utils.js

```diff
@@ -253,6 +253,9 @@ async function importModuleDynamicallyCallback(referrerSymbol, specifier, phase,
   // and fall back to the default loader.
   if (referrerSymbol === vm_dynamic_import_main_context_default) {
     emitExperimentalWarning('vm.USE_MAIN_CONTEXT_DEFAULT_LOADER');
+    // TODO: Discover why this is needed for vm to error with the correct code when esm/defaultLoad
+    // is sync. @see https://github.com/nodejs/node/pull/57419#discussion_r2027158688
+    if (referrerName === 'node:internal/process/task_queues') { referrerName = undefined; }
```

Comment:
I won't block because I don't have time to look into module loading stuff for the time being, but landing with this  still looks rather icky.

Reply:
I didn't ask you to review so I could ignore your feedback 😉

Reply:
I was asked to approve it on Slack, though I guess being asked to approve != being asked to review :)

Reply:
Truuuue. Your input is appreciated though.

Reply:
Btw CI is green; I could have merged hours ago. I was waiting 🙂